File: python/feature-envy/good/rectangle.py
import math

class Rectangle:
    def __init__(self, width: float, height: float):
        self._width = width
        self._height = height

    def get_width(self) -> float:
        return self._width

    def get_height(self) -> float:
        return self._height

    def calculate_area(self) -> float:
        return self._width * self._height

    def calculate_perimeter(self) -> float:
        return 2 * (self._width + self._height)

    def is_square(self) -> bool:
        return self._width == self._height

    def calculate_diagonal(self) -> float:
        return math.sqrt(self._width ** 2 + self._height ** 2)

    def get_aspect_ratio(self) -> float:
        return self._width / self._height


class GeometryUtils:
    # Utility methods that don't belong to Rectangle can stay here
    @staticmethod
    def calculate_distance_between_points(x1: float, y1: float, x2: float, y2: float) -> float:
        return math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)

    @staticmethod
    def calculate_angle(opposite: float, adjacent: float) -> float:
        return math.atan(opposite / adjacent)

File: java/feature-envy/good/Rectangle.java
public class Rectangle {
    private double width;
    private double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    public double getWidth() {
        return width;
    }

    public double getHeight() {
        return height;
    }

    public double calculateArea() {
        return width * height;
    }

    public double calculatePerimeter() {
        return 2 * (width + height);
    }

    public boolean isSquare() {
        return width == height;
    }

    public double calculateDiagonal() {
        return Math.sqrt(width * width + height * height);
    }

    public double getAspectRatio() {
        return width / height;
    }
}

class GeometryUtils {
    // Utility methods that don't belong to Rectangle can stay here
    public static double calculateDistanceBetweenPoints(double x1, double y1, double x2, double y2) {
        return Math.sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
    }

    public static double calculateAngle(double opposite, double adjacent) {
        return Math.atan(opposite / adjacent);
    }
}
